src/Todo.tsx

```diff
@@ -1,45 +1,40 @@
 import React, { useEffect, useState } from 'react';
-import CreateList from './CreateList';
+import DisplayList from './DisplayList';
+import formatList from './formatList';
 
-export async function GetTodoList() {
-    const response = await fetch('http://127.0.0.1:8000/todo-list/')
-
-    if (!response.ok){
-        throw new Error(
-            `This is an HTTP error: The status is ${response.status}`
-        )
-    }
-    const actualData = response.json()
-    return actualData
-}
-
-function Todo() {
+function Todo(props: {fetchList: any, addTodo: any}) {
     const [data, setData] = useState(Array<any>);
+    const [todo, setTodo] = useState('');
     const [error, setError] = useState('');
 
-
     useEffect(() => {
 
-        async function Wrapper() {
-            const actualData = await GetTodoList()
-            console.log(actualData)
-            try {
-                setData(actualData)
-                setError('') 
-            } catch (error) {
-                setError('There is an error!!!!')
-            }
+        async function getData() {
+            const taskArr = await props.fetchList()
+            const actualData = formatList(taskArr)
+            setData(actualData)
         }
-        Wrapper()
+        getData()
     }, [])
 
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setTodo(event.target.value)
+    }
+
+    const handleClick = (event: any) => {
+        event.preventDefault()
+        setData([...data, todo])
+        props.addTodo(todo)
```

Comment:
If `props.addTodo` is making a web request, what happens if this errors? We talked a little about this, but you should really make the add call, and then update the data with the new result list. That might mean doing the add, checking the response, and then doing a get if the add doesn't return the updated todo list.

Reply:
Agreed. Will be implementing this feature